packages/mobile/src/identity/contactMapping.ts

```diff
@@ -105,164 +93,65 @@ function* updateRecipientsCache(
-  // we see errors from web3. So we break things down and limit parallelization
-  // This is still not perfect, errors due still occur randomly for some chunks
-  const numberChunks = chunk(newE164Numbers, MAPPING_CHUNK_SIZE)
-  const requestChunks = chunk(numberChunks, NUM_PARALLEL_REQUESTS)
-  Logger.debug(
-    TAG,
-    `Lookup up: ${numberChunks.length} number chunks across ${requestChunks.length} request rounds`
-  )
-  for (const requestChunk of requestChunks) {
-    yield all(
-      requestChunk.map((numberChunk) =>
-        call(fetchAndStoreAddressMappings, attestationsWrapper, numberChunk)
-      )
-    )
-  }
-
-  // Now that mappings are updated, update the recipient objects
-  // TODO(Rossy) Consider revisiting the use of addresses in recip objects (to avoid confusion with the maps)
-  const updatedE164NumberToAddress: E164NumberToAddressType = yield select(
-    e164NumberToAddressSelector
-  )
-  for (const newNumber of newE164Numbers) {
-    e164NumberToRecipients[newNumber].address = updatedE164NumberToAddress[newNumber] || undefined
-  }
-
-  yield call(updateRecipientsCache, e164NumberToRecipients, otherRecipients)
-}
-
-async function getAddresses(e164Numbers: string[], attestationsWrapper: AttestationsWrapper) {
-  Logger.debug(TAG, `Get addresses for ${e164Numbers.length} phone numbers`)
-  const phoneHashes = e164Numbers.map((phoneNumber) => getPhoneHash(phoneNumber))
-  const results = await attestationsWrapper.lookupPhoneNumbers(phoneHashes)
-  if (!results) {
-    return null
-  }
-
-  const addresses: Array<string | null> = []
-  for (const hash of phoneHashes) {
-    if (results[hash]) {
-      // TODO(Rossy) Add support for handling multiple addresses per number
-      const addressArray = Object.keys(results[hash])
-      const address = addressArray[addressArray.length - 1]
-      addresses.push(address.toLowerCase())
-    } else {
-      addresses.push(null)
-    }
-  }
-
-  if (!addresses || addresses.length !== e164Numbers.length) {
-    throw new Error('Address lookup length did not match numbers list length')
-  }
-
-  return addresses
-}
-
-const isValidAddress = (address: string) =>
-  typeof address === 'string' && !new BigNumber(address).isZero()
-
-export function* fetchAndStoreAddressMappings(
-  attestationsWrapper: AttestationsWrapper,
-  e164Numbers: string[],
-  incrementSyncProgress = true
-) {
+export function* fetchPhoneAddresses({ e164Number }: FetchPhoneAddressesAction) {
   try {
-    Logger.debug(TAG, `Fetch and store address mapping for ${e164Numbers.length} phone numbers`)
-
-    const addresses: Array<string | null> = yield call(retryAsync, getAddresses, 3, [
-      e164Numbers,
-      attestationsWrapper,
+    Logger.debug(TAG + '@fetchPhoneAddresses', `Fetching addresses for number`)
+    // Clear existing entries for those numbers so our mapping consumers
+    // know new status is pending.
+    yield put(updateE164PhoneNumberAddresses({ [e164Number]: undefined }, {}))
```

Comment:
i'm a little confused how this works but seems this action updates the e164Number object.

why is this not fetching the salt before getting Attestations with updateE164PhoneNumberSalts?

Reply:
See the comment above :)
This clears the cached phone number so we can check it again. No need to clear salt caches since salts never change.

Reply:
oh i see -- you mean the comments that I've literally highlighted and failed to read :P

ya that makes sense now